javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -561,20 +566,22 @@ private Vector(int indexShift, HashArrayMappedTrie<Integer, T> trie) {
 
     @Override
     public Vector<T> append(T element) {
-        if (isEmpty()) {
-            return of(element);
-        } else {
-            return new Vector<>(indexShift, trie.put(length() + indexShift, element));
-        }
+        return wrap(trie.appendLeaf(element));
     }
 
     @Override
     public Vector<T> appendAll(Iterable<? extends T> elements) {
-        HashArrayMappedTrie<Integer, T> result = trie;
-        for (T element : elements) {
-            result = result.put(result.size() + indexShift, element);
+        Objects.requireNonNull(elements, "elements is null");
+        if (isEmpty()) {
+            return ofAll(elements);
+        } else {
+            Vector<T> result = this;
+            for (T element : elements) {
+                result = result.append(element);
+            }
+            return result.isEmpty() ? empty()
+                                    : result;
         }
-        return new Vector<>(indexShift, result);
```

Comment:
sure, `_1` is an `Array`, `_2` is a `Vector` (https://github.com/javaslang/javaslang/pull/1504/files/a07dbff7904d85c9d5c7a457dc913d88a681ddff..dcc87bc47d9541d2e07e96cf42f174c23211a932#diff-c89f0d7cf85723aceb8e16223092d4f1R168)